src/plugins/discover/public/application/main/components/layout/use_discover_histogram.ts

```diff
@@ -292,37 +273,41 @@ export const useDiscoverHistogram = ({
   // Trigger a unified histogram refetch when savedSearchFetch$ is triggered
   useEffect(() => {
     const subscription = savedSearchFetch$.subscribe(() => {
-      if (isInitialized && !skipRefetch.current) {
-        input$.next({ type: 'refetch' });
+      if (!skipRefetch.current) {
+        unifiedHistogram?.refetch();
       }
+
       skipRefetch.current = false;
     });
 
     return () => {
       subscription.unsubscribe();
     };
-  }, [input$, isInitialized, savedSearchFetch$]);
-
-  // Don't render the unified histogram layout until initialized
-  return isInitialized
-    ? {
-        query,
-        filters,
-        timeRange,
-        topPanelHeight,
-        request,
-        hits,
-        chart,
-        breakdown,
-        disableAutoFetching: true,
-        input$,
-        onEditVisualization: canVisualize ? onEditVisualization : undefined,
-        onTopPanelHeightChange,
-        onChartHiddenChange,
-        onTimeIntervalChange,
-        onBreakdownFieldChange,
-        onTotalHitsChange,
-        onChartLoad,
-      }
-    : undefined;
+  }, [savedSearchFetch$, unifiedHistogram]);
+
+  return { hideChart, setUnifiedHistogramApi };
+};
+
+const createStateSyncObservable = (state$?: Observable<UnifiedHistogramState>) => {
+  return state$?.pipe(
+    map(({ lensRequestAdapter, chartHidden, timeInterval, breakdownField }) => ({
+      lensRequestAdapter,
+      chartHidden,
+      timeInterval,
+      breakdownField,
+    })),
+    distinctUntilChanged((prev, curr) => {
+      const { lensRequestAdapter: prevLensRequestAdapter, ...prevRest } = prev;
+      const { lensRequestAdapter: currLensRequestAdapter, ...currRest } = curr;
+
+      return prevLensRequestAdapter === currLensRequestAdapter && isEqual(prevRest, currRest);
+    })
+  );
+};
+
+const createTotalHitsObservable = (state$?: Observable<UnifiedHistogramState>) => {
+  return state$?.pipe(
+    map((state) => ({ status: state.totalHitsStatus, result: state.totalHitsResult })),
+    distinctUntilChanged((prev, curr) => prev.status === curr.status && prev.result === curr.result)
```

Comment:
you can probably use `_.isEqual` here.

Reply:
In this case `.result` may be an Error object, so I avoided using `isEqual` in favour of referential equality.